crypto/s2n_hash.c

```diff
@@ -505,7 +505,6 @@ int s2n_hash_new(struct s2n_hash_state *state)
 S2N_RESULT s2n_hash_state_validate(struct s2n_hash_state *state)
 {
     ENSURE_REF(state);
-    ENSURE_REF(state->hash_impl);
```

Comment:
Does this weaken the verification for the hash methods?

Reply:
No, validate functions should only contain properties that will be true across all s2n_hash functions, and that wasn't the case. We also have a GitHub issue https://github.com/awslabs/private-s2n-cbmc/issues/18 to improve these validity functions.